packages/esm-bed-management-app/src/bed-administration/form/add-edit-bed.workspace.tsx

```diff
@@ -0,0 +1,407 @@
+import React, { useEffect } from 'react';
+import {
+  type DefaultWorkspaceProps,
+  ResponsiveWrapper,
+  useLayoutType,
+  useSession,
+  showSnackbar,
+} from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import { Controller, useForm } from 'react-hook-form';
+import styles from './add-edit-bed.workspace.scss';
+import {
+  ButtonSet,
+  Button,
+  InlineLoading,
+  TextInput,
+  NumberInput,
+  TextArea,
+  Select,
+  SelectItem,
+  ComboBox,
+  Form,
+  MultiSelect,
+  Stack,
+  Tag,
+} from '@carbon/react';
+import classNames from 'classnames';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import capitalize from 'lodash-es/capitalize';
+import { useBedTags, useLocationsWithAdmissionTag } from '../../summary/summary.resource';
+import { type InitialData } from '../../types';
+import { editBed, saveBed, useBedType } from './add-edit-bed.resource';
+
+type AddEditBedWorkspaceProps = DefaultWorkspaceProps & {
+  bed?: InitialData;
+  mutateBeds: () => void;
+  defaultLocation?: { display: string; uuid: string };
+};
+
+const AddEditBedWorkspace: React.FC<AddEditBedWorkspaceProps> = ({
+  closeWorkspace,
+  closeWorkspaceWithSavedChanges,
+  promptBeforeClosing,
+  bed,
+  mutateBeds,
+  defaultLocation,
+}) => {
+  const { t } = useTranslation();
+  const session = useSession();
+  const locationUuid = session?.sessionLocation?.uuid;
+  const isTablet = useLayoutType() === 'tablet';
+
+  const { admissionLocations } = useLocationsWithAdmissionTag();
+  const { bedTypes } = useBedType();
+  const { bedTags } = useBedTags();
+
+  const numberInString = z.string().transform((val, ctx) => {
+    const parsed = parseInt(val);
+    if (isNaN(parsed) || parsed < 1) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: t('validNumberRequired', 'Please enter a valid number'),
+      });
+      return z.NEVER;
+    }
+    return val;
+  });
+
+  const bedFormSchema = z.object({
+    bedId: z
```

Comment:
```suggestion
    bedNumber: z
```

Can we change this field name to `bedNumber` to align with the backend?